addons/account/models/account_reconcile_model.py

```diff
@@ -238,79 +217,67 @@ def _get_write_off_move_lines_dict(self, st_line, move_lines=None):
-            if self.force_tax_included:
-                tax = tax[0].with_context(force_price_include=True)
-            new_aml_dicts += self._get_taxes_move_lines_dict(tax, writeoff_line)
-
-        # Second write-off line.
-        if self.has_second_line and self.second_account_id:
-            remaining_balance = balance - sum(aml['debit'] - aml['credit'] for aml in new_aml_dicts)
-            if self.second_amount_type == 'percentage':
-                line_balance = remaining_balance * (self.second_amount / 100.0)
-            elif self.second_amount_type == "regex":
-                match = re.search(self.second_amount_from_label_regex, st_line.name)
+        if residual_balance is None:
+            residual_balance = balance
+        for line in self.line_ids:
+            if not line.account_id or float_is_zero(residual_balance, precision_rounding=line_currency.rounding):
+                continue
+
+            if line.amount_type == 'percentage':
+                line_balance = residual_balance * (line.amount / 100.0)
+            elif line.amount_type == "regex":
+                match = re.search(line.amount_from_label_regex, st_line.name)
                 if match:
-                    line_balance = copysign(float(re.sub(r'\D' + self.decimal_separator, '', match.group(1)).replace(self.decimal_separator, '.')), remaining_balance)
+                    line_balance = copysign(float(re.sub(r'\D' + line.decimal_separator, '', match.group(1)).replace(line.decimal_separator, '.')) * (1 if balance > 0.0 else -1), balance)
                 else:
                     line_balance = 0
             else:
-                line_balance = self.second_amount * (1 if remaining_balance > 0.0 else -1)
-
-            second_writeoff_line = {
-                'name': self.second_label or st_line.name,
-                'account_id': self.second_account_id.id,
-                'analytic_account_id': self.second_analytic_account_id.id,
-                'analytic_tag_ids': [(6, 0, self.second_analytic_tag_ids.ids)],
+                line_balance = line.amount * (1 if residual_balance > 0.0 else -1)
+            writeoff_line = {
+                'name': line.label or st_line.name,
+                'account_id': line.account_id.id,
+                'analytic_account_id': line.analytic_account_id.id,
+                'analytic_tag_ids': [(6, 0, line.analytic_tag_ids.ids)],
                 'debit': line_balance > 0 and line_balance or 0,
                 'credit': line_balance < 0 and -line_balance or 0,
                 'reconcile_model_id': self.id,
             }
-            new_aml_dicts.append(second_writeoff_line)
-
-            if self.second_tax_ids:
-                second_writeoff_line['tax_ids'] = [(6, None, self.second_tax_ids.ids)]
-                tax = self.second_tax_ids
+            new_aml_dicts.append(writeoff_line)
+            if for_widget:
+                writeoff_line['to_check'] = line.model_id.to_check
```

Comment:
why only if for_widget==True, and not always?

Reply:
There is a hack because the field `to_check` is actually on the `account.move` in the database, but we have to put it virtually in the `account.move.line` in the reconciliation widget because of the way the reconciliation models work.